src/system/SystemPacketBuffer.cpp

```diff
@@ -508,45 +508,63 @@ void PacketBuffer::AddRef()
 
 PacketBufferHandle PacketBufferHandle::New(size_t aAvailableSize, uint16_t aReservedSize)
 {
-    // Adding three 16-bit-int sized numbers together will never overflow
-    // assuming int is at least 32 bits.
+    // Ensure that SIZE_MAX is at least 32 bits
```

Comment:
Why do we care about ensuring this?  Where do we use this information?

As far as I can tell, the INT_MAX and SIZE_MAX asserts are not checking anything we care about anymore....

Reply:
This info should be in the comment. Otherwise the comment just restatest the code.

There should be an explanation why each of these assumptions are required/necessary for our code to work correctly.

Reply:
we also seem to use uint64_t now for additions ... do we ever care about size_t sizing now?

Reply:
The uint64_t cast was to ensure that the addition result does not overflow given the max sizes of the corresponding addends. The check for size_t was to ensure that the cast back to a size_t fits. I have a run-time check for CanCastTo<size_t>, so that makes this static_assert for SIZE_MAX redundant. Have removed it.